### ISSUE TITLE
Regression in Groovy: ClassCastException for closures not returning Boolean

### ISSUE THREAD

**Issue by pkubowicz** [awaitility/awaitility#71]
Code:
```
Awaitility.await().until({ "something" })
```
1.7.0: passes
2.0.0: fails with
```
java.lang.String cannot be cast to java.lang.Boolean
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
	at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:100)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.run(ConditionAwaiter.java:215)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

AwaitilityGroovyBridge used `runnable as Callable<Boolean>` while AwaitilityExtensionModule uses `runnable as Callable`.

Implementing `AwaitilityTrait` does not help, it seems that the extension module takes precedence. You need to change the closure to `{ "something" != null }`.

[Release Notes](https://github.com/awaitility/awaitility/wiki/ReleaseNotes20#release-notes-200) don't mention any breaking changes, so I consider this as a regression.

Also, please update examples from https://github.com/awaitility/awaitility/wiki/Groovy - they use deprecated classes.

**Comment by johanhaleby** [comment]
Thanks for pointing this out, I didn't know this. Is there a way to work-around this in the code to make it work again that you're aware of?

**Comment by pkubowicz** [comment]
I tried writing my own extension module, but I gave up, as a special entry in the JAR manifest is needed.

**Comment by johanhaleby** [comment]
Ok, so the `org.codehaus.groovy.runtime.ExtensionModule` file in /META-INF/services is not enough?

**Comment by pkubowicz** [comment]
It is probably enough, but I would need to create a JAR with my new extensions and add a dependency on it to tests. In my case it was faster to change the code.

**Comment by johanhaleby** [comment]
I've published a new snapshot that I would be glad if you could try out to see if it works: `2.0.1-SNAPSHOT`.

You'll need the following maven repo:

```xml
<repositories>
        <repository>
            <id>sonatype</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
            <snapshots />
        </repository>
</repositories>
```

**Comment by pkubowicz** [comment]
This does not work, but I have a fix in my fork, I'll submit a merge request soon.

**Comment by szpak** [comment]
Related to #65.

Btw, @pkubowicz it is not required to build and add as a dependency a JAR to create a new extension. Putting a proper file in `META-INF/services` in `src/test/resources` should be enough.

**Comment by pkubowicz** [comment]
I submitted my changes in #72

**Comment by johanhaleby** [comment]
Thanks! Can I close this issue? (perhaps you could add the wiki updates as a new issue)

**Comment by johanhaleby** [comment]
I've also published a new snapshot

**Comment by pkubowicz** [comment]
I found some problems if you had `AwaitilityTrait` in your code. My fixes in #73.

**Comment by pkubowicz** [comment]
I don't see any problems with the current snapshot - I am closing this issue.
